pkg/kube/inject/inject.go

```diff
@@ -398,8 +403,13 @@ func RunTemplate(params InjectionParameters) (mergedPod *corev1.Pod, templatePod
 	if err != nil {
 		return nil, nil, err
 	}
+	var proxyUID, proxyGID int64
 
-	proxyUID, proxyGID := GetProxyIDs(params.namespace)
+	if params.namespace != nil {
```

Comment:
Not sure if it's in the scope of this PR, but we should also adapt the lines below so the namespace client is not created:

https://github.com/maistra/istio/blob/f1c1c5d98915b343c06f10b189eac6b24e74de23/pkg/kube/inject/webhook.go#L205-L209

Reply:
`GetProxyIDs()` already checks for `namespace != nil`. Maybe we should call `GetProxyIDsFromPod()` in it?

Reply:
hmm.. that would mean changing the func signature to include the `corev1.Pod` as well as the `corev1.Namespace`. so we would need to change both the func and the code block that's calling it. I think it's simpler this way